processmanager/types.go

```diff
@@ -85,6 +95,14 @@ type ProcessManager struct {
 	// executable. It caches results based on iNode number and device ID. Locked LRU.
 	elfInfoCache *lru.LRU[util.OnDiskFileIdentifier, elfInfo]
 
+	// frameCache stores mappings from BPF frame to the symbolized frames.
+	// This allows avoiding the overhead of re-doing user-mode symbolization
+	// of frames that we have recently seen already.
+	frameCache *lru.SyncedLRU[frameCacheKey, libpf.Frames]
```

Comment:
I'm thinking to make this just `lru.LRU`. That is drop the `Synced` portion of it. And document (the previously established requirement) that `HandleTrace` is called only from single goroutine concurrently. This should speed up things a bit.